[Yang] ip

Shawn532 opened the pull request:
# Duke
> Success is not an accident, success is actually a choice. -Stephen Curry

DukePro frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,
1. download it from [here](https://github.com/Shawn532/ip/releases).
2. double-click it.
3. add your tasks.
4. let it manage yours tasks for you :+1: :yum:

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] GUI (coming soon)
- [ ] Reminders (coming soon)

Commit:
Add Gradle support

Commit:
level-1

Commit:
level-1

Commit:
level-2

Commit:
why no unmark???

Commit:
level-4

Commit:
level-6

Commit:
level-7

Commit:
test

Commit:
fix the delete method

Commit:
change the listAllItems method

Commit:
implement a basic structure of writing into a text file

Commit:
Implements the basic structure of level-8

Commit:
implements the basic structure of level-8

Commit:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Deadline.java

Commit:
add 4 classes

Commit:
Implements TaskList

Commit:
implements Storage class

Commit:
Fix the bug of the path

Commit:
Clears most of the Duke class to TaskList class

Commit:
Add packages

Commit:
Adds java-doc

Commit:
Level-9 find

Commit:
Merge remote-tracking branch 'upstream/add-gradle-support'

Commit:
Updates the gradle file.

brendantwh commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,69 @@
+    private Ui ui;
+    private Storage storage;
+
+    /**
+     * Initializes the Duke chatbot.
+     * @param filePath Specifies the path to save the content.
+     */
+
+    public Duke(String filePath) {
+        ui = new Ui();
+        tasks = new TaskList();
+        storage = new Storage(filePath);
+    }
```
Blank line between the Javadocs comment and the method declaration.

brendantwh commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,33 @@
+public class Storage {
+    private String filePath;
+
+    /**
+     * A constructor to specify the path
+     * @param filePath the path to save the file and load the file from
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Updates the file
+     * @param str The strings to be written in the file
+     */
```
Nitpicking here, but there should be a blank line between description and the param section.

brendantwh reviewed (commented):
Couldn't find much coding standard violations, apart from the minor ones mentioned. Code seemed pretty clean and Javadocs for relevant methods were concise. Good job!

stevenlimhw commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,69 @@
+        while (true) {
+            String input = ui.readCommand();
+            String command = input.split(" ")[0];
+            switch (command) {
```
Perhaps, the commands may be better represented using enums to ensure that only commands included in the enum is considered as a valid command. Another idea would be to create classes for the specific types of commands.

stevenlimhw commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,25 @@
+        try {
+            this.exactTime = LocalDate.parse(eventTime);
+        } catch (DateTimeParseException e) {
+            System.out.println("Wrong input format!");
```
The error message can be made clearer here. Perhaps, you can show how the correct input format is really like.

stevenlimhw commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,143 @@
+    public void addATask(String item) throws DukeException {
+        String[] arr = item.split(" ", 2);
+        String type = arr[0];
+        switch (type) {
```
This is not a code quality violation, but rather a coding standard violation. In CS2103T, if I remember correctly, the cases should not have an additional indent with respect to the switch line.

stevenlimhw commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,25 @@
+import java.time.format.DateTimeParseException;
+
+public class Deadline extends Task {
+    public String dl;
```
The naming of this field is quite unclear. Perhaps, you can try naming it with more descriptive words instead of just letters.

stevenlimhw commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,25 @@
+
+public class Deadline extends Task {
+    public String dl;
+    public LocalDate exactTime;
```
For the sake of information hiding (an OOP principle), fields of the class should be private. In the case when we need to access the field data, we can simply create a getter method that returns the data.

Commit:
Completes the GUI tutorial.

Commit:
Uses FXML

Commit:
Trying to implement getResponse

Commit:
Divides command into many classes.

Commit:
Implements the Parser class

Commit:
Implements the basic functions of GUI

Commit:
Adds a yml file

Commit:
Updates a message.

Commit:
Add junit

Commit:
Change the ui

Commit:
Update the user interface to normal situation.

Commit:
Use ui to print the message.

Commit:
Add Main.java.

Commit:
Add assertions

Commit:
Improves the code quality

Shawn532 committed:
Merge pull request #2 from Shawn532/branch-A-Assertions

Add assertions

Shawn532 committed:
Merge pull request #3 from Shawn532/branch-A-CodeQuality

Improves the code quality

Commit:
Add a brief UG and a picture of the chatbox.

The user guide provides a basic understanding of how the chatbox works. The picture shows the todo feature, mark feature and list feature.

You can have a look at the user guide at https://shawn532.github.io/ip/.

Commit:
no message

Commit:
Implements the ReminderCommand class

Commit:
Loads the task from the tasks.txt file.

Commit:
Implements the loading function and writing function.

Commit:
Fix bugs of the duke chatbox. Bye command will terminate the program now.  An empty todo command will not cause error.

Commit:
Updates the UG

Commit:
Change the format of the output message.

Commit:
Fix the bug of writing into the text file when input if invalid.

Now when the input format in invalid. It will say can not parse, wrong input format instead of terminating.

Commit:
no message
